ec/ecresource/deploymentresource/elasticsearch_remote_cluster_expanders.go

```diff
@@ -0,0 +1,82 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package deploymentresource
+
+import (
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/api/deploymentapi/esremoteclustersapi"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/cloud-sdk-go/pkg/util/ec"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func handleRemoteClusters(d *schema.ResourceData, client *api.API) error {
+	if keyIsEmptyUnchanged(d, "elasticsearch.0.remote_cluster") {
+		return nil
+	}
+
+	res, err := expandRemoteClusters(
+		d.Get("elasticsearch.0.remote_cluster").([]interface{}),
+	)
+	if err != nil {
+		return err
+	}
+
+	return esremoteclustersapi.Update(esremoteclustersapi.UpdateParams{
+		API:             client,
+		DeploymentID:    d.Id(),
+		RefID:           d.Get("elasticsearch.0.ref_id").(string),
+		RemoteResources: res,
+	})
+}
+
+func expandRemoteClusters(raw []interface{}) (*models.RemoteResources, error) {
```

Comment:
We don't seem to be returning errors anywhere 🤔 It seems safe to only have `*models.RemoteResources` as a return value on the signature?